File: dashboard/src/exchangeRate.js
import { writable } from 'svelte/store';

export const eurRate = writable(null);

export async function fetchEurRate() {
  try {
    const r = await fetch('/api/exchange-rate');
    if (!r.ok) return;
    const data = await r.json();
    eurRate.set(data.eur ?? null);
  } catch {
    eurRate.set(null);
  }
}

File: backend/api/exchange_rate.py
import json
import urllib.request
from fastapi import APIRouter

router = APIRouter()

_EXCHANGE_RATE_URL = "https://open.er-api.com/v6/latest/USD"


@router.get("/api/exchange-rate")
def get_exchange_rate():
    try:
        with urllib.request.urlopen(_EXCHANGE_RATE_URL, timeout=5) as resp:
            data = json.loads(resp.read())
        rate = data.get("rates", {}).get("EUR")
        return {"eur": float(rate) if rate is not None else None}
    except Exception:
        return {"eur": None}
